### DISCUSSION TITLE
If I don't pass "Content-Type" in headers, what does it send by default?

### DISCUSSION THREAD

**Discussion by feng-jianwei** [curl/curl#12810] [Q&A] [answered]
curl_easy_setopt(m_curl, CURLOPT_HTTPHEADER，m_curlPtrSList)；
if no "Content-Type" in m_curlPtrSList, what does it send by default?

libcurl version: 4.7.0

**Comment by dfandrich** [comment]
You need to give more details. If the program doesn't set a Content-Type:
header, libcurl won't send one, unless there's one implied such as when
using CURLOPT_POSTFIELDS.

**Reply by feng-jianwei** [reply]
What I want to know is the general scenario (when a get post delete etc request is sent)
For a request with a body, libcurl automatically fills in the body format. If there is no body, libcurl does not actively populate Content-Type. Do I understand that right?

**Reply by bagder** [reply, marked as answer]
The documentation for CURLOPT_POSTFIELDS says "This POST is a normal application/x-www-form-urlencoded kind (and libcurl sets that Content-Type by default when this option is used),"

**Reply by feng-jianwei** [reply]
I found it.Thanks reply
